### PR TITLE
Delete path.h

### PR DESCRIPTION
**Pull request by snnn**
### Description
Delete path.h and replace all occurrences of onnxruntime::Path with std::filesystem::path. 
Previously we couldn't use C++17's std::filesystem because it was not supported in iOS 12(which was released in 2018). Now we dropped the support for iOS 12. 

### Motivation and Context
To simplify code.  For example, if an EP wants to use the Path class, now it can directly use it without going through a wrapper. And the standard implementation can handle various path types better. (We didn't take much consideration on UNC path, "/" as a path separator on Windows,  etc).

### CODE REVIEW

**Review by mszhanyi — APPROVED** [review]

File: onnxruntime/test/debug_node_inputs_outputs/debug_node_inputs_outputs_utils_test.cc [hunk 1]
```diff
@@ -27,7 +27,7 @@ void VerifyTensorProtoFileData(const PathString& tensor_proto_path, gsl::span<co
 
   std::vector<T> actual_data{};
   actual_data.resize(expected_data.size());
-  ASSERT_STATUS_OK(utils::UnpackTensor(tensor_proto, Path{}, actual_data.data(), actual_data.size()));
+  ASSERT_STATUS_OK(utils::UnpackTensor(tensor_proto, std::filesystem::path(), actual_data.data(), actual_data.size()));
```
Comment (mszhanyi) [inline onnxruntime/test/debug_node_inputs_outputs/debug_node_inputs_outputs_utils_test.cc:30]: std::filesysstem::path(tensor_proto_path) ?

File: onnxruntime/test/debug_node_inputs_outputs/debug_node_inputs_outputs_utils_test.cc [hunk 2]
```diff
@@ -48,7 +48,7 @@ void VerifyTensorProtoFileDataInt4(const PathString& tensor_proto_path,
 
   std::vector<Int4x2Base<Signed>> actual_data{};
   actual_data.resize(expected_data.size());
-  ASSERT_STATUS_OK(utils::UnpackTensor(tensor_proto, Path{}, actual_data.data(), num_elems));
+  ASSERT_STATUS_OK(utils::UnpackTensor(tensor_proto, std::filesystem::path(), actual_data.data(), num_elems));
```
Comment (mszhanyi) [inline onnxruntime/test/debug_node_inputs_outputs/debug_node_inputs_outputs_utils_test.cc:51]: std::filesysstem::path(tensor_proto_path) ?

File: onnxruntime/test/debug_node_inputs_outputs/debug_node_inputs_outputs_utils_test.cc (hunk 1 above)
Comment (snnn) [reply onnxruntime/test/debug_node_inputs_outputs/debug_node_inputs_outputs_utils_test.cc:30]: The path is model path, where the model file lives. Here we do not have a model.

File: onnxruntime/test/debug_node_inputs_outputs/debug_node_inputs_outputs_utils_test.cc (hunk 2 above)
Comment (snnn) [reply onnxruntime/test/debug_node_inputs_outputs/debug_node_inputs_outputs_utils_test.cc:51]: The path is model path, where the model file lives. Here we do not have a model.

**Review by mszhanyi — APPROVED** [review]

**Pull request merged**